packages/core/src/typings/openapi.ts

```diff
@@ -202,46 +230,61 @@ export type Oas3ParameterStyle =
   | 'pipeDelimited'
   | 'deepObject';
 
-export interface Oas3RequestBody {
+export interface Oas3RequestBody<T extends Oas3Schema | Oas3_1Schema = Oas3Schema | Oas3_1Schema> {
   description?: string;
   required?: boolean;
-  content: { [mime: string]: Oas3MediaType };
+  content: { [mime: string]: Oas3MediaType<T> };
 }
 
-export interface Oas3Responses {
-  [code: string]: Oas3Response;
+export interface Oas3Responses<T extends Oas3Schema | Oas3_1Schema> {
```

Comment:
Let's add a default for backward compatibility:

```suggestion
export interface Oas3Responses<T extends Oas3Schema | Oas3_1Schema = Oas3Schema | Oas3_1Schema> {
```